src/corecel/io/detail/UniqueFileNamer.cc

```diff
@@ -0,0 +1,54 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/io/detail/UniqueFileNamer.cc
+//---------------------------------------------------------------------------//
+#include "UniqueFileNamer.hh"
+
+#include <chrono>
+#include <iomanip>
+#include <sstream>
+
+#include "corecel/Assert.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with original filename.
+ */
+UniqueFileNamer::UniqueFileNamer(std::string const& filename)
+{
+    CELER_EXPECT(!filename.empty());
+    std::ostringstream stem;
+
+    if (auto pos = filename.find_last_of('.');
```

Comment:
Since we already use `std::filesystem` elsewhere we could use `std::filesystem::path::stem` and `std::filesystem::path::extension`

Reply:
Good call.  I'll change here and have add a note to follow-on to check for other uses in the codebase.